### PR TITLE
[Go] Complete methods exercise concept

### PR DESCRIPTION
**Pull request by oanaOM**
This is for #2049. Please review.

### CODE REVIEW

File: languages/go/concepts/methods/about.md
Comment (ErikSchierboom) [inline languages/go/concepts/methods/about.md:2]: We'll automatically add this header on the website

```suggestion
```

File: languages/go/concepts/methods/links.json
Comment (ErikSchierboom) [inline languages/go/concepts/methods/links.json:8]: Maybe use a user-friendly name here? Something like:

```suggestion
    "description": "Pointer receivers"
```

p.s. I don't know if the above suggestion makes any sense

File: languages/go/concepts/methods/links.json
Comment (ErikSchierboom) [inline languages/go/concepts/methods/links.json:12]: See comment above

File: languages/go/exercises/concept/methods/.docs/hints.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/methods/.docs/hints.md:3]:
```suggestion
- Add a field to the `Car` struct that keeps track of the driven distance
```

File: languages/go/exercises/concept/methods/.docs/hints.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/methods/.docs/hints.md:4]:
```suggestion
- Create a [method][method] that has a receiver of type `Car` and updates the driven distance
```

File: languages/go/exercises/concept/methods/.docs/hints.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/methods/.docs/hints.md:8]:
```suggestion
- Create a [method][method] that has a receiver of type `Car`
```

File: languages/go/exercises/concept/methods/.docs/hints.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/methods/.docs/hints.md:13]:
```suggestion
- Create a [method][method] that has a receiver of type `Car`
```

File: languages/go/exercises/concept/methods/.docs/introduction.md
Comment (ErikSchierboom) [inline languages/go/exercises/concept/methods/.docs/introduction.md:1]:
```suggestion
## methods
```

**Review by ErikSchierboom — APPROVED** [review]
Excellent work! Some minor comments

File: languages/go/concepts/methods/about.md
Comment (oanaOM) [reply languages/go/concepts/methods/about.md:2]: Cool

File: languages/go/concepts/methods/links.json
Comment (oanaOM) [reply languages/go/concepts/methods/links.json:8]: Makes sense.

**Comment by oanaOM** [comment]
Great. Thanks for you review @ErikSchierboom! I've added the suggestions in.

**Comment by ErikSchierboom** [comment]
@oanaOM Perfect! I'll let the Go maintainers take a look and merge this PR. Thanks a lot for doing this!

File: languages/go/concepts/methods/about.md
Comment (tehsphinx) [inline languages/go/concepts/methods/about.md:26]: Maybe remove all but the type declaration and the method? Could make it easier to understand.

File: languages/go/concepts/methods/about.md
Comment (tehsphinx) [inline languages/go/concepts/methods/about.md:29]:
```suggestion
Methods with a value receiver operate on a copy of the value passed to it, meaning that any modification done to the receiver inside the method is not visible to the caller.
```

File: languages/go/concepts/methods/about.md
Comment (tehsphinx) [inline languages/go/concepts/methods/about.md:49]: This example is not assigning any new values to the struct fields from within the method. Could be done without a pointer receiver. Can we adjust this?

File: languages/go/exercises/concept/methods/.docs/introduction.md
Comment (tehsphinx) [inline languages/go/exercises/concept/methods/.docs/introduction.md:1]: In this file the same comments apply as in the concept/methods/about.md file

**Review by tehsphinx — COMMENTED** [review]
Looks very good @oanaOM!

I left some comments. Feel free to apply or disagree 😏

File: languages/go/concepts/methods/about.md
Comment (oanaOM) [reply languages/go/concepts/methods/about.md]: Sure, I've removed them.

File: languages/go/concepts/methods/about.md
Comment (oanaOM) [reply languages/go/concepts/methods/about.md]: Absolutely, my bad. Please check my last commit .

File: languages/go/exercises/concept/methods/.docs/introduction.md
Comment (oanaOM) [reply languages/go/exercises/concept/methods/.docs/introduction.md:1]: Yes

File: languages/go/concepts/methods/about.md
Comment (tehsphinx) [reply languages/go/concepts/methods/about.md]: Hmm, this still works the same if you remove the pointer receiver.

Here an example that would not work any more if you remove the receiver pointer:

```go
type rect struct {
	width, height int
}

func (r *rect) squareIt() {
	r.height = r.width
}

func main() {
	r := rect{width: 10, height: 20}
	fmt.Printf("Width: %d, Height: %d\n", r.width, r.height)
	// Output: Width: 10, Height: 20

	r.squareIt()
	fmt.Printf("Width: %d, Height: %d\n", r.width, r.height)
	// Output: Width: 10, Height: 10
}
```

Comment (oanaOM) [reply languages/go/concepts/methods/about.md]: @tehsphinx oh, I see now. Thanks for helping out with this!

**Comment by tehsphinx** [comment]
Thanks @oanaOM!!

**Pull request merged**
